site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -25,49 +28,78 @@
 @Service
 public class SiteServiceImpl implements SiteService {
 
+    private final Map<String, SiteConfiguration> sitesByType = new HashMap<>();
+    private final Map<String, String> parentTypeByChildType = new HashMap<>();
     private final EntityStore<Site, String> siteStore;
     private final ModelEntityValidation<Site> entityValidation;
     private final Logger logger = LoggerFactory.getLogger(SiteServiceImpl.class);
 
     /**
+     * @param configurationProvider provide Site Configuration
      * @param siteStore        Site store interface
      * @param entityValidation Site validation interface
      */
     @Autowired
-    public SiteServiceImpl(final EntityStore<Site, String> siteStore,
+    public SiteServiceImpl(SiteConfigurationProvider configurationProvider,
+                           final EntityStore<Site, String> siteStore,
                            ModelEntityValidation<Site> entityValidation) {
+        if (configurationProvider == null) {
+            throw new InitialisationError("site configuration provider cannot be null");
+        }
         if (siteStore == null) {
             throw new InitialisationError("site store cannot be null");
         }
         if (entityValidation == null) {
             throw new InitialisationError("entity validation cannot be null");
         }
+        var configuration = configurationProvider.getConfiguration();
         this.siteStore = siteStore;
         this.entityValidation = entityValidation;
+        addTypesToMap(configuration);
         logger.info(Services.STARTUP.message());
     }
 
+    private void addTypesToMap(final SiteConfiguration configuration) {
+        this.sitesByType.put(configuration.getType(), configuration);
+        if (configuration.getChild() != null) {
```

Comment:
could use CollectionsUtils library or similar to do null check to make code tidier..e.g emptyIfNull(collection).foreach

Reply:
Used CollectionsUtils.emptyIfNull(collection)